tests/test_algorithm.py

```diff
@@ -231,22 +231,6 @@ def fake_method(*args, **kwargs):
             with ZiplineAPI(algo):
                 self.assertIs(sentinel, getattr(zipline.api, name)())
 
-    def test_cannot_iterate_over_data(self):
```

Comment:
Should we just invert this test instead of deleting? i.e. ensure that data can be iterated over, (and possibly check for warning?)